### PR TITLE
Core 598

### PR DESCRIPTION
**Pull request by mobileguru007**
Finished Rock, Paper, Scissors game on React Native

### CODE REVIEW

**Review by jeapostrophe — COMMENTED** [review]
Same comment as with the other one.

**Comment by mobileguru007** [comment]
Updated gitignore files

File: examples/tut-10/App.js [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+import { Platform } from 'react-native';
+
+const reach = loadStdlib(process.env);
+
+const handToInt = { ROCK: 0, PAPER: 1, SCISSORS: 2 };
+const intToOutcome = ['Bob wins!', 'Draw!', 'Alice wins!'];
+const { standardUnit } = reach;
+const defaults = { defaultFundAmt: '10', defaultWager: '0.005', standardUnit };
+
+class App extends React.Component {
```
Comment (hamirmahal) [inline examples/tut-10/App.js:25]: How come we're using class-based components instead of functional components?

Comment (jeapostrophe) [reply examples/tut-10/App.js:25]: No idea. @DanBurton wrote it :)

File: examples/tut-10/Makefile [hunk 2]
```diff
@@ -0,0 +1,5 @@
+.PHONY: run
+run:
+	@echo 'Please use `reach react` to run this DApp'
```
Comment (jeapostrophe) [inline examples/tut-10/Makefile:3]: Is this true?

File: examples/tut-10/__tests__/App-test.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
+/**
```
Comment (jeapostrophe) [inline examples/tut-10/__tests__/App-test.js:1]: Is this needed? It doesn't look like it does anything

File: examples/tut-10/algoMAW.patch [hunk 4]
```diff
@@ -0,0 +1,13 @@
+--- examples/tut-9/index.js	2021-08-31 12:43:24.801893800 -0400
```
Comment (jeapostrophe) [inline examples/tut-10/algoMAW.patch:1]: Have you tested with these patches? I assume you have not and should remove them

File: examples/tut-10/reach [hunk 5]
```diff
@@ -0,0 +1,72 @@
+#!/bin/sh
```
Comment (jeapostrophe) [inline examples/tut-10/reach:1]: This file should not be committed

**Review by jeapostrophe — CHANGES_REQUESTED** [review]
Does the yarn lock need to be there?

File: examples/tut-10/App.js (hunk 1 above)
Comment (hamirmahal) [reply examples/tut-10/App.js:25]: Ha, I see. I was just curious: the syntax for functional components is much cleaner.

**Comment by hamirmahal** [comment]
@jeapostrophe Not sure if this is helpful, but `yarn.lock` tells people using `yarn` as their package manager what dependencies they need to install, in the same way that `package.json` and `package-lock.json` do that for `npm`.

File: examples/tut-10/App.js [hunk 6]
```diff
@@ -10,61 +10,59 @@ import React from 'react';
...
 
 class App extends React.Component {
   constructor(props) {
     super(props);
-    this.state = { view: 'ConnectAccount', ...defaults };
+    this.state = {view: 'ConnectAccount', ...defaults};
     const init = async () => {
       await reach.setProviderByName('TestNet');
-    }
+    };
```
Comment (hamirmahal) [inline examples/tut-10/App.js:30]: @mobileguru007 are you using the Eslint extension for Visual Studio Code? It'll catch missing semicolons for you automatically if your project is using `eslint` and you have the semicolon rule enabled.

File: examples/tut-10/App.js [hunk 7]
```diff
@@ -10,61 +10,59 @@ import React from 'react';
...
     init();
   }
   async componentDidMount() {
     if (await reach.canFundFromFaucet()) {
-      this.setState({ view: 'FundAccount' });
+      this.setState({view: 'FundAccount'});
     } else {
-      this.setState({ view: 'DeployerOrAttacher' });
+      this.setState({view: 'DeployerOrAttacher'});
```
Comment (hamirmahal) [inline examples/tut-10/App.js:37]: I recommend installing Prettier to automatically take care of spacing issues.

File: examples/tut-10/App.js [hunk 8]
```diff
@@ -10,61 +10,59 @@ import React from 'react';
...
 
   async setAccount(isDeployer = true) {
     // const acc = await reach.getDefaultAccount();
-    const phrase = isDeployer ?
-      "humor sting race bonus unit arctic speak fine wood double hip crouch"
-      :
-      "oppose settle table giggle flush seven addict wrap pull jelly payment purchase";
+    const phrase = isDeployer
+      ? 'humor sting race bonus unit arctic speak fine wood double hip crouch'
+      : 'oppose settle table giggle flush seven addict wrap pull jelly payment purchase';
```
Comment (hamirmahal) [inline examples/tut-10/App.js:45]: Prettier will also handle formatting of code like this for you, automatically.

File: examples/tut-10/App.js (hunk 6 above)
Comment (hamirmahal) [reply examples/tut-10/App.js:30]: I just realized 3f5c4a6 has the commit message `Eslinting Done`, so we are using `eslint` here.

I think that'll be helpful to us long-term.

**Comment by mobileguru007** [comment]
> Does the yarn lock need to be there?

**Pull request closed without merging**